Kernel/Scheduler.cpp

```diff
@@ -449,94 +449,67 @@ bool Scheduler::donate_to(Thread* beneficiary, const char* reason)
 #ifdef SCHEDULER_DEBUG
     dbgprintf("%s(%u:%u) donating %u ticks to %s(%u:%u), reason=%s\n", Process::current->name().characters(), Process::current->pid(), Thread::current->tid(), ticks_to_donate, beneficiary->process().name().characters(), beneficiary->pid(), beneficiary->tid(), reason);
 #endif
-    context_switch(*beneficiary);
+
     beneficiary->set_ticks_left(ticks_to_donate);
-    switch_now();
-    return false;
+    return yield();
 }
 
 bool Scheduler::yield()
 {
     InterruptDisabler disabler;
     ASSERT(Thread::current);
-    if (!pick_next())
-        return false;
-    switch_now();
-    return true;
-}
-
-void Scheduler::pick_next_and_switch_now()
-{
-    bool someone_wants_to_run = pick_next();
-    ASSERT(someone_wants_to_run);
-    switch_now();
-}
-
-void Scheduler::switch_now()
-{
-    Descriptor& descriptor = get_gdt_entry(Thread::current->selector());
-    descriptor.type = 9;
-    asm("sti\n"
-        "ljmp *(%%eax)\n" ::"a"(&Thread::current->far_ptr()));
-}
 
-bool Scheduler::context_switch(Thread& thread)
-{
-    thread.set_ticks_left(time_slice_for(thread));
-    thread.did_schedule();
+    Thread* swapping_from = Thread::current;
+    Thread* swapping_to = pick_next();
```

Comment:
"Swapping" generally means something else in OS lingo (swapping memory to disk.)
Some alternatives:
* `switching_from` and `switching_to`
* `incoming_thread` and `outgoing_thread`